pycloudlib/azure/instance.py

```diff
@@ -2,19 +2,42 @@
 """Azure instance."""
 
 import time
+from collections import namedtuple
+from enum import Enum
 from typing import List, Optional
 
+import requests
+from azure.core.exceptions import ResourceExistsError
+
 from pycloudlib.errors import PycloudlibTimeoutError
 from pycloudlib.instance import BaseInstance
 
+BootDiagnostics = namedtuple("BootDiagnostics", ["console_log_url", "logs"])
+
+
+class VMInstanceStatus(Enum):
+    """Represents Azure VM state."""
+
+    UNPROVISIONED = 0
+    FAILED_PROVISION = 1
+    ACTIVE = 2
+    DELETED = 3
```

Comment:
Can be simplified with `auto()` since the underlying value of the enumeration in irrelevant.